// react-kirk-pc-ui/webpack.config.js
const path = require('path');
const webpack = require('webpack');

module.exports = {
    entry: './app/component/lib/index.js',
    output: {
        path: path.resolve('dist'),
        filename: 'index.js',
        library: 'react-kirk-pc-ui',
        libraryTarget: 'umd',
    },
    module: {
        rules: [
            {
                test: /\.jsx?$/,
                loader: 'babel-loader',
                include: [
                    path.resolve(__dirname, "app/component/lib")
                ]
            }
        ]
    },
    externals: {
        react: {
            root: 'React',
            var: 'React',
            commonjs: 'react',
            commonjs2: 'react',
            amd: 'react',
        },
        'react-dom': "this react-dom"
    },
    plugins: [
        /*压缩js代码*/
        new webpack.optimize.UglifyJsPlugin({
            output: {
                comments: false/*删除版权信息*/
            },
            compress: {
                warnings: false
            }
        }),
    ],
    resolve: {
        extensions: [".js", ".jsx"]
    }
};
